pkg/apply/task/inv_add_task.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package task
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+	"sigs.k8s.io/cli-utils/pkg/apply/event"
+	"sigs.k8s.io/cli-utils/pkg/apply/taskrunner"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// InvAddTask encapsulates structures necessary to add/merge inventory
+// into the cluster. The InvAddTask should add/merge inventory references
+// before the actual object is applied.
+type InvAddTask struct {
+	InvClient inventory.InventoryClient
+	InvInfo   inventory.InventoryInfo
+	Objects   []*unstructured.Unstructured
+}
+
+func (i *InvAddTask) Name() string {
+	return "inventory-add"
```

Comment:
We might want to have unique names here in cases where we have more than one instance in the task queue. The idea was that clients can use the name to identify tasks/groups for the `ActionGroup` events.